django/views/i18n.py

```diff
@@ -318,3 +331,120 @@ def json_catalog(request, domain='djangojs', packages=None):
         'plural': plural,
     }
     return http.JsonResponse(data)
+
+
+class JavascriptCatalog(View):
+    """
+    Returns the selected language catalog as a javascript library.
+
+    Receives the list of packages to check for translations in the
+    kwargs `packages` key either from an infodict or as a +-delimited
+    string from the request. Default is 'django.conf'.
+
+    Additionally you can override the gettext domain for this view,
+    but usually you don't want to do that, as JavaScript messages
+    go to the djangojs domain. But this might be needed if you
+    deliver your JavaScript source from Django templates.
+    """
+    domain = 'djangojs'
+    packages = None
+
+    def get(self, request, *args, **kwargs):
+        locale = get_language()
+        domain = kwargs.get('domain', self.domain)
+        # If packages are not provided, default to all installed packages, as
+        # DjangoTranslation without localedirs is harvesting them all.
+        packages = kwargs.get('packages', '').split('+') or self.packages
+        paths = self.get_paths(packages) if packages else None
```

Comment:
So the js18n view for admin, why does it still load django.conf  on top of django.contrib.admin? Is it there because no one ever noticed there are no longer js ranslations for django.conf? If so it would be good to have it cleared also






De: Claude Paroz
Enviado el: ‎miércoles‎, ‎09‎ de ‎marzo‎ de ‎2016 ‎19‎:‎21
Para: django/django
CC: Cristiano Coelho





In django/views/i18n.py:
> +
> +    Additionally you can override the gettext domain for this view,
> +    but usually you don't want to do that, as JavaScript messages
> +    go to the djangojs domain. But this might be needed if you
> +    deliver your JavaScript source from Django templates.
> +    """
> +    domain = 'djangojs'
> +    packages = None
> +
> +    def get(self, request, *args, **kwargs):
> +        locale = get_language()
> +        domain = kwargs.get('domain', self.domain)
> +        # If packages are not provided, default to all installed packages, as
> +        # DjangoTranslation without localedirs is harvesting them all.
> +        packages = kwargs.get('packages', '').split('+') or self.packages
> +        paths = self.get_paths(packages) if packages else None


Maybe django.conf used to contain a djangojs.po in the past? Now it does not so I don't see the point in including it. And yes, I think we should encourage the practice to have one well-cached jsi18n view instead of several per-app views.

—
Reply to this email directly or view it on GitHub.